api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -53,6 +54,9 @@ message UpstreamTlsContext {
   //
   // Defaults to 1, setting this to 0 disables session resumption.
   google.protobuf.UInt32Value max_session_keys = 4;
+
+  // If true, the handshake will fail if the keyUsage extension is present and incompatible with the TLS usage.
```

Comment:
Is this using a WKT because we eventually want to change the default to true? Or will the default always be false? Either way I would specify the default is false.

Additionally, it would be optimal if we could warn somehow if using a certificate where this would fail and the setting is false? Or potentially warn if the certs are OK and this could be set to true?

Reply:
Exactly. As i mentioned in the PR description, the default will be changed to true eventually. But yes, I should note it in the comment. Done.

Regarding warning, there is a flag `was_key_usage_invalid` that will be set to TRUE [here](https://github.com/google/boringssl/blob/80a243e07ef77156af66efa7d22ac35aba44c1b3/ssl/handshake_client.cc#L1399) when certificate would fail but the error was skipped because the setting is false. So, we can set the enforcement to true when we don't see this flag to ever flipped to true